File: 7CM4 - DESARROLLO DE APLICACIONES MOVILES/Parcial 1/practicas/Practica 6/app/src/main/java/com/example/intentosjava/ResultActivity.java
package com.example.intentosjava;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

public class ResultActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result);

        TextView textViewResult = (TextView) findViewById(R.id.textViewResult);

        double a = getIntent().getDoubleExtra("a", 0);
        double b = getIntent().getDoubleExtra("b", 0);
        double c = getIntent().getDoubleExtra("c", 0);
        double discriminante = b * b - 4 * a * c;

        String resultado;
        if (discriminante > 0) {
            double x1 = (-b + Math.sqrt(discriminante)) / (2 * a);
            double x2 = (-b - Math.sqrt(discriminante)) / (2 * a);
            resultado = "a = " + a + ", b = " + b + ", c = " + c
                    + "\n\nDos raíces reales:\nx1 = " + x1 + "\nx2 = " + x2;
        } else if (discriminante == 0) {
            double x = -b / (2 * a);
            resultado = "a = " + a + ", b = " + b + ", c = " + c
                    + "\n\nRaíz doble:\nx = " + x;
        } else {
            // Ejercicio 1: raíces complejas con notación a + jb.
            double parteReal = -b / (2 * a);
            double parteImaginaria = Math.abs(Math.sqrt(-discriminante) / (2 * a));
            resultado = "a = " + a + ", b = " + b + ", c = " + c
                    + "\n\nRaíces complejas:\nx1 = " + parteReal + " + " + parteImaginaria + "j"
                    + "\nx2 = " + parteReal + " - " + parteImaginaria + "j";
        }

        textViewResult.setText(resultado);
    }
}

File: 7CM4 - DESARROLLO DE APLICACIONES MOVILES/Parcial 1/practicas/Practica 7/app/src/main/java/com/example/intentoskotlin/ResultActivity.kt
package com.example.intentoskotlin

import android.app.Activity
import android.os.Bundle
import android.widget.TextView
import kotlin.math.abs
import kotlin.math.sqrt

class ResultActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_result)

        val coefA = intent.getDoubleExtra("coefA", 0.0)
        val coefB = intent.getDoubleExtra("coefB", 0.0)
        val coefC = intent.getDoubleExtra("coefC", 0.0)

        val discriminante = coefB * coefB - 4 * coefA * coefC
        val resultText = findViewById<TextView>(R.id.resultText)

        val resultado = if (discriminante > 0) {
            val x1 = (-coefB + sqrt(discriminante)) / (2 * coefA)
            val x2 = (-coefB - sqrt(discriminante)) / (2 * coefA)
            "Las raíces son:\nx1 = ${"%.3f".format(x1)}\nx2 = ${"%.3f".format(x2)}"
        } else if (discriminante == 0.0) {
            val x = -coefB / (2 * coefA)
            "La raíz doble es:\nx = ${"%.3f".format(x)}"
        } else {
            val parteReal = -coefB / (2 * coefA)
            val parteImaginaria = abs(sqrt(-discriminante) / (2 * coefA))
            "Raíces complejas:\nx1 = ${"%.3f".format(parteReal)} + ${"%.3f".format(parteImaginaria)}j" +
                "\nx2 = ${"%.3f".format(parteReal)} - ${"%.3f".format(parteImaginaria)}j"
        }

        resultText.text = resultado
    }
}
